source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -39,6 +39,32 @@ struct ExtProcFilterStats {
   ALL_EXT_PROC_FILTER_STATS(GENERATE_COUNTER_STRUCT)
 };
 
+inline constexpr absl::string_view ExtProcGrpcStatsName = "ext-proc-filter-state";
+
+class ExtProcGrpcStats : public Envoy::StreamInfo::FilterState::Object {
```

Comment:
(optional / subjective): IIUC this is named ExtProcGrpcStats to distinguish it from ExtProcFilterStats, and because all current values in it are gRPC-related. This makes sense and certainly works.

But, ISTM that the more fundamental difference between this new class and ExtProcFilterStats is that the stats in this new class are per-stream. So another possible name for this class would be ExtProcPerStreamStats, which would give more leeway in the future for other fields. For example, if we decide that we want to capture per-stream failure messages or status flags to be included in BDN ULS logging, that would go in this class as well, but would not be strictly gRPC-related.